src/state/schedule-data/foundation-info/foundation-info.reducer.ts

```diff
@@ -3,33 +3,18 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 import * as _ from "lodash";
-import { FoundationInfoModel } from "./foundation-info.model";
-import { FoundationInfoAction, FoundationInfoActionType } from "./foundation-info.action-creator";
-import { scheduleDataInitialState } from "../schedule-data-initial-state";
+import { createReducer } from "@reduxjs/toolkit";
 import {
-  createActionName,
-  isScheduleAction,
-  ScheduleActionModel,
-  ScheduleActionType,
-} from "../schedule.actions";
+  FoundationInfoAction,
+  updateChildrenAndExtraworkers,
+} from "./foundation-info.action-creator";
+import { scheduleDataInitialState } from "../schedule-data-initial-state";
+import { addNewSchedule, isScheduleAction, updateSchedule } from "../schedule.actions";
 
-export function foundationInfoReducerF(name: string) {
-  return (
-    state: FoundationInfoModel = scheduleDataInitialState.month_info,
-    action: ScheduleActionModel | FoundationInfoAction
-  ): FoundationInfoModel => {
-    switch (action.type) {
-      case createActionName(name, ScheduleActionType.ADD_NEW):
-      case createActionName(name, ScheduleActionType.UPDATE):
-        if (!isScheduleAction(action)) {
-          return state;
-        }
-        const monthInfo = action.payload?.month_info;
-        if (!monthInfo) {
-          return state;
-        }
-        return { ...monthInfo };
-      case createActionName(name, FoundationInfoActionType.UPDATE_CHILDREN_AND_EXTRAWORKERS):
+export const foundationInfoReducerF = (name: string) =>
+  createReducer(scheduleDataInitialState.month_info, (builder) => {
+    builder
+      .addCase(updateChildrenAndExtraworkers, (state, action) => {
```

Comment:
Wcześniej ta akcja była uzależniona od tego do którego grafiku my wprowadzaliśmy zmianę: tymczasowego czy aktualnego.
To pozwalało zrobić te dwie wersje w miarę niezależnymi od siebie.
Teraz tak nie jest:
https://youtu.be/LKJtc3zJUbU